src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenLocalFunctionTests.cs

```diff
@@ -2100,7 +2100,12 @@ public void OptionalParameterCodeGen_Reflection()
         var p2 = d.Method.GetParameters();
         Console.WriteLine(p2[0].HasDefaultValue);
         Console.WriteLine(p2[0].DefaultValue);", @"True
-5", new[] { "System" });
+5", usings: "System", diagnostics: new[]
+            {
+                // (8,17): warning CS9069: Parameter 1 has default value '5' in method group and '<missing>' in the target delegate type.
+                //         var d = (Action<int>)TestAction;
+                Diagnostic(ErrorCode.WRN_OptionalParamValueMismatch, "(Action<int>)TestAction").WithArguments("1", "5", "<missing>").WithLocation(8, 17)
```

Comment:
I understand you called it out in the description, but this doesn't match the speclet, and there should be no warning for this case. The main case we want to warn for method group conversions is when both sides have a default value, and they don't match.